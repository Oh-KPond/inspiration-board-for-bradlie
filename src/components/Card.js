import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  handleSubmit = (event) => {
    event.preventDefault();

    this.props.deleteCardCallback(this.props.id);
  }

  render() {
    let displayText;
    let displayEmoji;

    if (this.props.text) {
      displayText = <h3 className="card__content-text">{this.props.text}</h3>;
    }

    if (this.props.emoji) {
      displayEmoji = <p className="card__content-emoji">{emoji.getUnicode(this.props.emoji)}</p>;
    }


    return (
      <div className="card">
        <article className="card__content">
          {displayText}
          {displayEmoji}
        </article>
        <form className="card__delete" onSubmit={this.handleSubmit}>
          <button className="button" type="submit">x</button>
        </form>
      </div>
    )
  }
}

Card.propTypes = {
  text: PropTypes.string,
  emoji: PropTypes.string,
  id: PropTypes.number,
  deleteCardCallback: PropTypes.func,
};

export default Card;
