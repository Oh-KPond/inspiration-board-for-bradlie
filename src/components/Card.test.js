import React from 'react';
import { mount, shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    // Mount the component in the DOM
    // Given
    const wrapper = shallow(<Card />);
    // When and Then
    expect(wrapper).toMatchSnapshot();
    // Remove component from the DOM
    wrapper.unmount();
  });
});
