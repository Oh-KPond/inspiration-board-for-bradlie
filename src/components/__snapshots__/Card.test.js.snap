// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="card__content"
>
          
</article>,
        <form
          className="card__delete"
          onSubmit={[Function]}
>
          <button
                    className="button"
                    type="submit"
          >
                    x
          </button>
</form>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="button"
            type="submit"
>
            x
</button>,
          "className": "card__delete",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "className": "button",
            "type": "submit",
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="card__content"
>
            
</article>,
          <form
            className="card__delete"
            onSubmit={[Function]}
>
            <button
                        className="button"
                        type="submit"
            >
                        x
            </button>
</form>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
            ],
            "className": "card__content",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button"
              type="submit"
>
              x
</button>,
            "className": "card__delete",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "x",
              "className": "button",
              "type": "submit",
            },
            "ref": null,
            "rendered": "x",
            "type": "button",
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
